src/main/kotlin/org/jetbrains/bio/big/BigWig.kt

```diff
@@ -64,7 +72,15 @@ class BigWigFile private constructor(input: RomBuffer,
                                query: ChromosomeInterval,
                                overlaps: Boolean): Sequence<WigSection> {
         val chrom = chromosomes[query.chromIx]
-        return sequenceOf(input.with(dataOffset, dataSize, compression) {
+        var localRomBuf = lastRomBuf
+
+        if (localRomBuf.first.match(chrom, dataOffset, dataSize)) {
+            localRomBuf = Pair(RomBufferState(chrom, dataOffset, dataSize),
+                              input.decompress(dataOffset, dataSize, compression))
+            lastRomBuf = localRomBuf
+        }
+
+        return sequenceOf(localRomBuf.second!!.duplicate().with {
             val chromIx = getInt()
             assert(chromIx == query.chromIx) { "section contains wrong chromosome" }
```

Comment:
Also I'd replaced `assert` with `check` so that it works even without `-ea` option.